Upgrade the python version to 3.10

twunderlich-grapl opened the pull request:
<!-- Thank you for your contribution to Grapl! Please take some time
to fill out this short template to help us review and understand your
PR. -->

### Which issue does this PR correspond to?
Fixes https://github.com/grapl-security/issue-tracker/issues/974

<!-- Please provide a link to the GitHub Issue this pull request is
meant to address. -->

### What changes does this PR make to Grapl? Why?
This upgrades python to 3.10
<!-- Please describe the functional impact of these changes and the
rationale behind them. This will help us understand your approach to
solving the problem. -->

### How were these changes tested?
Automated tests
<!-- Please describe how these changes were tested. There should be
sufficient detail that reviewers can replicate your testing
procedure. If the procedure is a manual one that's OK, your
description will help us work with you to get it automated. -->

Commit:
Upgrade the python version to 3.10

christophermaier commented on src/python/grapl_analyzerlib/mypy.ini:
```diff
@@ -1,10 +1,10 @@
 [mypy]
-python_version = 3.8
+python_version = 3.10
```
I'm not sure this file is even used anymore 🤔

(This can be ironed out outside of this PR, though.)

christophermaier commented on .python-version:
```diff
@@ -1 +1 @@
-3.8.13
+3.10.5
```
Note that we'll need to ensure this version of Python is present on our build agents: https://github.com/grapl-security/grapl-infrastructure/blob/e9e86c60b18fd5f152d87c75d19079847f65b784/packer/buildkite-base-ami.pkr.hcl#L197-L205

Commit:
Switch to 3.10.4

Pyenv doesn't support 3.10.5 yet, so we should use 3.10.4

Commit:
Upgrade to 3.10.5 since pyenv 2.3.2 supports it

Commit:
Merge remote-tracking branch 'origin/main' into twunderlich/upgrade-to-python3.10

Commit:
Update builddocs script for pants python not code python

Since we now have a difference between the python version for pants
and the python version for the code, we're now having to be explicit
about which one we use. This sucks

Commit:
Merge branch 'main' into twunderlich/upgrade-to-python3.10

Commit:
Upgrade lockfiles for python 3.10

Commit:
Add a python version supported by Pants to container

Since we're using pants inside the container we need
2 versions of python, our python 3.10 and a python
that's supported by pants.
